base_layer/core/src/chain_storage/lmdb_db/lmdb.rs

```diff
@@ -75,6 +75,18 @@ where
         .map_err(|e| ChainStorageError::AccessError(e.to_string()))
 }
 
+pub fn lmdb_replace<K, V>(txn: &WriteTransaction, db: &Database, key: &K, val: &V) -> Result<(), ChainStorageError>
+where
+    K: Serialize,
+    V: Serialize,
+{
+    let key_buf = serialize(key)?;
+    let val_buf = serialize(val)?;
+    txn.access()
+        .put(&db, &key_buf, &val_buf, put::Flags::empty())
```

Comment:
Curious.. What was the error when `lmdb_insert` was called on an existing key?

Reply:
It produced a "duplicate key detected" error.